network/transport/stcp_socket.c

```diff
@@ -0,0 +1,260 @@
+#include <stdlib.h>
+#include <string.h>
+#include "socket.h"
+#include "socket_interface.h"
+#include "error.h"
+#include "util.h"
+#include "trng.h"
+#include "mbedtls/ssl.h"
+#include "mbedtls/entropy.h"
+
+/**
+ * @struct stcp_socket_desc
+ * @brief Fields used by socket
+ */
+struct stcp_sockets_desc {
+	/** TCP Socket reference */
+	struct socket_desc		*sock;
+	/** True random number generator reference */
+	struct trng_desc		*trng;
+	/* Mbed structures */
+	/** Certification authority certificate */
+	mbedtls_x509_crt		ca_cert[1];
```

Comment:
why use array of size 1 ?

Reply:
All mbedtls function work with a reference to this structures. This is a workaround to avoid using `&ldesc->ca_cert` in all function calls.

Instead of:
```
mbedtls_x509_crt ca_cert;
mbedtls_x509_crt_init(&ldesc->ca_cert);
```
To write:
```
mbedtls_x509_crt ca_cert[1];
mbedtls_x509_crt_init(ldesc->ca_cert);
```

Reply:
i do not like this workaround i would prefer &ldesc->ca_cert. If the requested parameter is a pointer it makes more sense to call it with &